fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/DefaultHavingApiGenerator.java

```diff
@@ -44,6 +43,15 @@ public DefaultHavingApiGenerator(ConfigurationLoader loader) {
         this.metricDictionary = loader.getMetricDictionary();
     }
 
+    /**
+     * Constructor.
+     *
+     * @param metricDictionary  Dictionary of metrics for generating APIs
+     */
+    public DefaultHavingApiGenerator(Map<String, LogicalMetric> metricDictionary) {
```

Comment:
I assume the other constructor above this constructs a global metric dictionary, right? If it's the case, it's a good idea to add a paragraph of description to clarify that(and this constructor as well)